google/cloud/internal/timer_queue.cc

```diff
@@ -104,27 +113,30 @@ void TimerQueue::Service() {
     // This may run user code (in the continuations for the future). That may
     // do all kinds of things, including calling back into this class to create
     // new timers. We cannot hold the mutex while it is running.
-    p.set_value(until);
+    p.set_value(key.first);
```

Comment:
The test expectations can now be made exact.

Reply:
Done.